### PR TITLE
Add org.texstudio.TeXstudio

### PR DESCRIPTION
**Pull request by Alexander-Wilms**

### CODE REVIEW

File: org.texstudio.TeXstudio.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+app-id: org.texstudio.TeXstudio
```
Comment (barthalion) [inline org.texstudio.TeXstudio.yaml]: File doesn't follow style guide: https://github.com/flathub/flathub/wiki/YAML-Style-Guide

File: org.texstudio.TeXstudio.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  - "echo \"I\nn\" | TEXLIVE_INSTALL_PREFIX=/app ./install-tl -portable"
+  - yes | rm -r /app/texmf-dist/doc
+- name: openjpeg
+  buildsystem: cmake
+  sources:
+  - type: archive
+    url: https://github.com/uclouvain/openjpeg/archive/v2.3.0.tar.gz
+    sha256: 3dc787c1bb6023ba846c2a0d9b1f6e179f1cd255172bde9eb75b01f1e6c7d71a
+- name: poppler
+  buildsystem: cmake
```
Comment (barthalion) [inline org.texstudio.TeXstudio.yaml]: cmake-ninja?

File: org.texstudio.TeXstudio.yaml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  - type: file
+    url: http://mirrors.ctan.org/systems/texlive/Images/texlive.iso
+    sha512: 7b7f0dd0eab3bfffe52c5cd1139c7f75d029b9ff4c4ce0e57e06834705522f4ec0c02cd99a80b053c6619abda51c20a60f8e91e91781bdc2b9b60fc2e5708adb
+  build-commands:
+  - 7z x texlive.iso
+  - chmod +x install-tl
+  - "echo \"I\nn\" | TEXLIVE_INSTALL_PREFIX=/app ./install-tl -portable"
+  - yes | rm -r /app/texmf-dist/doc
+- name: openjpeg
+  buildsystem: cmake
```
Comment (barthalion) [inline org.texstudio.TeXstudio.yaml]: cmake-ninja?

**Comment by barthalion** [comment]
bot, build org.texstudio.TeXstudio

File: org.texstudio.TeXstudio.yaml (hunk 1 above)
Comment (Alexander-Wilms) [reply org.texstudio.TeXstudio.yaml]: Done

File: org.texstudio.TeXstudio.yaml (hunk 2 above)
Comment (Alexander-Wilms) [reply org.texstudio.TeXstudio.yaml]: Done

File: org.texstudio.TeXstudio.yaml (hunk 3 above)
Comment (Alexander-Wilms) [reply org.texstudio.TeXstudio.yaml]: Done

**Comment by Alexander-Wilms** [comment]
The builbot had issues:

x86_64
```
Dependency Sdk: org.kde.Sdk 5.9
FB: Running: flatpak --user info org.kde.Sdk/x86_64/5.9 --show-commit
FB: Running: flatpak --user update --subpath= org.kde.Sdk/x86_64/5.9
Looking for updates...
error: GPG signatures found, but none are in trusted keyring
Error running (null): Child process exited with code 1
```

i386:
```
Dependency Sdk: org.kde.Sdk 5.9
FB: Running: flatpak --user info org.kde.Sdk/i386/5.9 --show-commit
FB: Running: flatpak --user update --subpath= org.kde.Sdk/i386/5.9
Looking for updates...
Warning: Can't find dependencies: Unable to load metadata from remote flathub: unknown error
Error updating remote metadata for 'flathub': GPG signatures found, but none are in trusted keyring
Error: Failed to update org.kde.Sdk/i386/5.9: Unable to load summary from remote flathub: GPG signatures found, but none are in trusted keyring
error: One or more operations failed
Error running (null): Child process exited with code 1
```

**Comment by Alexander-Wilms** [comment]
A download timed out on i386 and TeXlive failed to build for some reason on aarch64.

**Comment by TingPing** [comment]
bot, build org.texstudio.TeXstudio

**Comment by Alexander-Wilms** [comment]
While I can compile newly created documents (stored in /tmp), I can't yet compile documents stored in ~.

Theres two issue:

* latex/lualatex/pdflatex etc. don't have read access to included (via \include or \input) .tex files
* Compiling documents eben without included files fails due to not being able to write .log, .aux etc files

--filesystem=home doesn't help, since the path TeXstudio sees after opening a file is still /run/user/[...]

I remember one issue being opened against flatpak regarding paths being rewritten, but I can't find it right now. Does anybody know of a way to grant an application complete access to ~ without any portals?

**Comment by swilmet** [comment]
Yes I've faced similar issues with the paths to files that change, which is problematic for a text editor where the directory is shown, see the [comment here](https://gitlab.gnome.org/GNOME/tepl/blob/e65aec28aa84c2fabf10ab9d9ef3750a6f2bb039/tepl/tepl-application-window.c#L165). The only solution that I found was to use `--filesystem=host` in the flatpak finish-args. But there is maybe a better solution.

**Comment by Alexander-Wilms** [comment]
Thanks for the info. When using --filesystem=home or host, I can simple use the bundled terminal to navigate to the directory containing the .tex file and run lualatex without any special options and it works just fine.

So the only issue is the fake path being returned by the portal. I don't know if there's a Qt equivalent of  GtkFileChooserDialog, but there was already some work being done regarding thisproblem:

https://github.com/flatpak/xdg-desktop-portal/issues/141

https://github.com/flatpak/xdg-desktop-portal/pull/148

Deleting /run/user/1000/flatpak-info fixes the issue, cf. https://github.com/flatpak/flatpak/issues/1276

**Comment by nedrichards** [comment]
bot, build org.texstudio.TeXstudio

**Comment by barthalion** [comment]
Is it blocked by issues with the portal?

**Comment by Alexander-Wilms** [comment]
Yes

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/org.texstudio.TeXstudio

**Comment by nedrichards** [comment]
@Alexander-Wilms anyone else we need to add?

**Comment by Alexander-Wilms** [comment]
Possibly @timhoffm

But since the issue with xdg-desktop-portal has not yet been resolved, maybe the repo should be deleted again for the time being.

**Comment by Alexander-Wilms** [comment]
The issue with the portal somehow resolved itself.

**Pull request closed without merging**